client/site-profiler/hooks/use-site-profiler-record-analytics.ts

```diff
@@ -0,0 +1,26 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { useEffect } from 'react';
+import type { CONVERSION_ACTION } from 'calypso/site-profiler/hooks/use-define-conversion-action';
+
+/**
+ * Record analytics events for site-profiler.tsx root element only
+ * The idea is to move all root element analytics events to this hook
+ */
+export default function useSiteProfilerRecordAnalytics(
+	domain: string,
+	isDomainValid?: boolean,
+	conversionAction?: CONVERSION_ACTION
+) {
+	useEffect( () => {
+		recordTracksEvent( 'calypso_site_profiler_page_view' );
+	}, [] );
+
+	useEffect( () => {
+		domain &&
+			isDomainValid &&
```

Comment:
There's also special domain cases that falls into isDomainValid = false, like:
- wordpress.com/site-profiler
- localhost
- 127.0.0.1

For these cases, should we also trigger this event as well?